river/InputManager.zig

```diff
@@ -138,6 +138,21 @@ pub fn inputAllowed(self: Self, wlr_surface: *wlr.Surface) bool {
         true;
 }
 
+/// Maps/Unmaps all input devices to/from their configured output.
+/// This is called on outputs being added or removed.
+pub fn reassignDevices(self: *Self) void {
+    for (self.configs.items) |conf| {
+        if (conf.@"map-to-output") |_| {
+            var dev_iter = self.devices.iterator(.forward);
+            while (dev_iter.next()) |dev| {
+                if (@import("globber").match(dev.identifier, conf.glob)) {
+                    conf.apply(dev);
+                }
+            }
+        }
+    }
+}
```

Comment:
This is incorrect as it will apply all configuration settings for configs with a non-null map-to-output field, not just the map-to-output setting, and potentially break the property that the config with the most specific matching glob takes precedence.

I think this function should not try to take shortcuts as an "optimization" and rather perform a full reconfiguration. That is, use the following code from `command/input.zig` and refactor that flie to use this function as well.
```zig
    var it = server.input_manager.devices.iterator(.forward);
    while (it.next()) |device| {
        for (server.input_manager.configs.items) |config| {
            if (globber.match(device.identifier, config.glob)) {
                config.apply(device);
            }
        }
    }
```